start-site/src/main/java/io/spring/start/site/extension/dependency/springazure/SpringAzureDefaultBuildCustomizer.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.start.site.extension.dependency.springazure;
+
+import io.spring.initializr.generator.buildsystem.Build;
+import io.spring.initializr.generator.spring.build.BuildCustomizer;
+
+/**
+ * A {@link BuildCustomizer} that automatically adds "spring-cloud-azure-starter" when any
+ * Spring Cloud Azure library is selected.
+ *
+ * @author Yonghui Ye
+ */
+public class SpringAzureDefaultBuildCustomizer implements BuildCustomizer<Build> {
+
+	@Override
+	public void customize(Build build) {
+		if (build.dependencies().items().anyMatch((u) -> u.getGroupId().equals("com.azure.spring")) && !build
+				.dependencies().items().anyMatch((u) -> u.getArtifactId().equals("spring-cloud-azure-starter"))) {
+			build.dependencies().add("azure-support");
```

Comment:
Why is this necessary? Every `spring-cloud-azure-starter-*` module should depend on `spring-cloud-azure-starter` just as each of Spring Boot's own starters depends on `spring-boot-starter`.

Reply:
Sorry, that's not quite what I meant. This whole customiser should not be necessary as each `spring-cloud-azure-starter-*` should declare a dependency on `spring-cloud-azure-starter` in its pom.

How would we get into a situation where there's a `com.azure.spring:*` dependency but no `spring-cloud-azure-starter-*` dependency?

Reply:
Thanks for your feedback. There really is no such scenario at the moment, `SpringAzureDefaultBuildCustomizer` is unnecessary.